### PR TITLE
add comments to errors

### PR DESCRIPTION
**Pull request by mingwli**
Consolidate comment into errors per discussion [Meeting Notes](https://teams.microsoft.com/l/entity/com.microsoft.teamspace.tab.wiki/tab::a59deabc-5855-48a0-8bf4-dcb59f36e1bc?context=%7B%22subEntityId%22%3A%22%7B%5C%22pageId%5C%22%3A3%2C%5C%22sectionId%5C%22%3A4%2C%5C%22origin%5C%22%3A2%7D%22%2C%22channelId%22%3A%2219%3A22aff63434ff4345bc5627fa74b9145b%40thread.skype%22%7D&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)

### CODE REVIEW

File: src/docfx/Errors.cs [hunk 1]
```diff
@@ -4,174 +4,390 @@
...
         public static Error YamlSyntaxError(in Range range, string message)
             => new Error(ErrorLevel.Error, "yaml-syntax-error", message, range: range);
 
+        /// <summary>
+        /// Used duplicate yaml key in markdown yml header or schema document(yml).
+        /// </summary>
         public static Error YamlDuplicateKey(in Range range, string key)
             => new Error(ErrorLevel.Error, "yaml-duplicate-key", $"Key '{key}' is already defined, remove the duplicate key.", range: range);
 
         public static Error InvalidYamlHeader(Document file, Exception ex)
```
Comment (superyyrrzz) [inline src/docfx/Errors.cs]: miss one?

File: src/docfx/Errors.cs [hunk 2]
```diff
@@ -4,174 +4,390 @@
...
 
+        /// <summary>
+        /// In yaml-format toc, topicHref should reference an article,
+        /// rather than relative path or another toc file.
+        /// </summary>
         public static Error InvalidTopicHref(Document relativeTo, string topicHref)
             => new Error(ErrorLevel.Error, "invalid-topic-href", $"The topic href '{topicHref}' can only reference to a local file or absolute path", relativeTo.ToString());
 
+        /// <summary>
+        /// In markdown-format toc, link(treated as inclusion) can only be toc file, folder or absolute path.
```
Comment (superyyrrzz) [inline src/docfx/Errors.cs]: minor: you can use upper case for SHOULD/CAN/ONLY, as most descriptions here are about "what is wrong", only `InvalidTopicHref` and `InvalidTocHref` are saying "what is right"

File: src/docfx/Errors.cs [hunk 3]
```diff
@@ -4,174 +4,390 @@
...
+        ///   - failed to download due to bad network
+        ///   - when update user profile cache fails, need to download verify etag
+        /// </summary>
         public static Error DownloadFailed(string url, string message)
             => new Error(ErrorLevel.Error, "download-failed", $"Download '{url}' failed: {message}");
 
+        /// <summary>
+        /// Failed to update user profile cache file.
+        /// Examples:
+        ///   - when config.github.UpdateRemoteUserCache is turned on, and docfx fails to
```
Comment (superyyrrzz) [inline src/docfx/Errors.cs]: use cref?

File: src/docfx/Errors.cs [hunk 4]
```diff
@@ -4,174 +4,390 @@
...
+        /// Can't find a file referenced by configuration, or user writes a non-existing link.
+        /// Examples:
+        ///   - define user_profile.json file in config, while the file doesn't exist
+        ///   - href referencing a non-existing file
+        /// </summary>
         public static Error FileNotFound(string relativeTo, string path)
             => new Error(ErrorLevel.Warning, "file-not-found", $"Cannot find file '{path}' relative to '{relativeTo}'", relativeTo);
 
+        /// <summary>
+        /// Failed to resolve uid defined by [link](xref:uid) syntax.
```
Comment (superyyrrzz) [inline src/docfx/Errors.cs]: or `<xref:uid>` syntax

File: src/docfx/Errors.cs [hunk 5]
```diff
@@ -4,174 +4,390 @@
...
+        /// </summary>
         public static Error UidNotFound(Document file, string uid, string rawXref)
             => new Error(ErrorLevel.Warning, "uid-not-found", $"Cannot find uid '{uid}' using xref '{rawXref}'", file.ToString());
 
+        /// <summary>
+        /// File contains git merge conflict.
```
Comment (superyyrrzz) [inline src/docfx/Errors.cs]: add example about this?

File: src/docfx/Errors.cs [hunk 6]
```diff
@@ -4,174 +4,390 @@
...
+        /// Inclusion is a redirections entry like [!INCLUDE[](redirect.md)]
+        /// </summary>
         public static Error IncludeRedirection(Document relativeTo, string path)
             => new Error(ErrorLevel.Warning, "include-is-redirection", $"Referenced inclusion {path} relative to '{relativeTo}' shouldn't belong to redirections", relativeTo.ToString());
 
+        /// <summary>
+        /// More than one files are resolved to the same output path.
+        /// Examples:
+        ///   - in config's redirections section, defined an entry key that's also a file in build scope
+        ///   - different file extension with same filename, like `Toc.yml and `Toc.md`
```
Comment (superyyrrzz) [inline src/docfx/Errors.cs]: miss ` after toc.yml

File: src/docfx/Errors.cs [hunk 7]
```diff
@@ -4,174 +4,390 @@
...
+        /// </summary>
         public static Error OutputPathConflict(string path, IEnumerable<Document> files)
             => new Error(ErrorLevel.Error, "output-path-conflict", $"Two or more files output to the same path '{path}': {Join(files, file => file.ContentType == ContentType.Redirection ? $"{file} <redirection>" : file.ToString())}");
 
+        /// <summary>
+        /// Multiple files defined in config's redirections are redirected to the same url,
```
Comment (superyyrrzz) [inline src/docfx/Errors.cs]: Does the condition need to **AND** "with `redirect_document_id` set to `true`" (I am not sure of the config name...)

File: src/docfx/Errors.cs [hunk 8]
```diff
@@ -4,174 +4,390 @@
...
             => new Error(ErrorLevel.Warning, "reserved-metadata", $"Metadata '{name}' is reserved by docfx, remove this metadata from {removeFrom}", null, range);
 
+        /// <summary>
+        /// Failed to compute specific info of a commit.
+        /// </summary>
         public static Error GitLogError(string repoPath, int errorCode)
             => new Error(ErrorLevel.Error, "git-log-error", $"Error computing git log [{errorCode}] for '{repoPath}', did you used a shadow clone?");
 
+        /// <summary>
+        /// Git isn't installed.
```
Comment (superyyrrzz) [inline src/docfx/Errors.cs]: minor: `git.exe` should be more specific

File: src/docfx/Errors.cs [hunk 9]
```diff
@@ -4,174 +4,390 @@
...
+        /// Examples:
+        ///   - non-loc build, docfx.yml/docfx.json doesn't exist at the repo root
+        ///   - loc build, docfx.yml/docfx.json doesn't exist at the neither loc repo root
+        ///     nor source repo root
+        /// </summary>
         public static Error ConfigNotFound(string docsetPath)
             => new Error(ErrorLevel.Error, "config-not-found", $"Cannot find 'docfx.yml/docfx.json' at '{docsetPath}'");
 
+        /// <summary>
+        /// Two files include each other.
```
Comment (live1206) [inline src/docfx/Errors.cs]: not only markdown inclusion, xref circular reference is using this error code as well

File: src/docfx/Errors.cs [hunk 10]
```diff
@@ -4,174 +4,390 @@
...
             => new Error(ErrorLevel.Warning, "unknown-field", $"Could not find member '{propName}' on object of type '{typeName}'.", range: range, jsonPath: path);
 
+        /// <summary>
+        /// Schema document with violate content type/value against predefined models(not syntax error).
+        /// </summary>
         public static Error ViolateSchema(in Range range, string message, string path)
             => new Error(ErrorLevel.Error, "violate-schema", $"{message}", range: range, jsonPath: path);
 
+        /// <summary>
+        /// Used unknow YamlMime.
```
Comment (superyyrrzz) [inline src/docfx/Errors.cs]: unknown

File: src/docfx/Errors.cs [hunk 11]
```diff
@@ -4,174 +4,390 @@
...
+        /// Config's monikerRange and monikerRange defined in yaml header has no intersection,
+        /// or moniker-zone defined in article.md has no intersection with file-level monikers.
+        /// </summary>
         public static Error EmptyMonikers(string message)
             => new Error(ErrorLevel.Warning, "empty-monikers", message);
 
+        /// <summary>
+        /// Referenced an article using uid with invalid moniker(?view=).
+        /// Examples:
+        ///   - article with uid `a` has only netcore-1.0 & netcore-1.1 version, but get referenced with @a? view = netcore - 2.0
```
Comment (superyyrrzz) [inline src/docfx/Errors.cs]: remove spaces in `@a? view = netcore - 2.0`?

File: src/docfx/Errors.cs (hunk 1 above)
Comment (mingwli) [reply src/docfx/Errors.cs]: this one is never used

File: src/docfx/Errors.cs [hunk 12]
```diff
@@ -4,174 +4,399 @@
...
         public static Error ConfigNotFound(string docsetPath)
             => new Error(ErrorLevel.Error, "config-not-found", $"Cannot find 'docfx.yml/docfx.json' at '{docsetPath}'");
 
+        /// <summary>
+        /// Two files include each other.
+        /// Examples:
+        ///   - a.md includes b.md and reversely
+        ///   - toc1 references toc2 and reversely
+        ///   - toc references itself
+        ///   - a.md references b.json's property with xref syntax, and b.json includes amd reversely
```
Comment (live1206) [inline src/docfx/Errors.cs]:
```suggestion
        ///   - a.md references b.json's property with xref syntax, and b.json includes a.md reversely
```

File: src/docfx/Errors.cs [hunk 13]
```diff
@@ -4,174 +4,399 @@
...
             => new Error(ErrorLevel.Error, "git-clone-failed", $"Cloning git repository '{url}' ({Join(branches)}) failed.");
 
+        /// <summary>
+        /// Yaml header defined in article.md isn't an object.
+        /// </summary>
         public static Error YamlHeaderNotObject(bool isArray)
             => new Error(ErrorLevel.Warning, "yaml-header-not-object", $"Expect yaml header to be an object, but got {(isArray ? "an array" : "a scalar")}");
 
+        /// <summary>
+        /// Syntax error in yaml file(not duplicate key).
```
Comment (mingwli) [inline src/docfx/Errors.cs]:
```suggestion
        /// Syntax error in yaml file(not duplicate key) .
```

**Pull request merged**